lib/mapObjects/CGTownInstance.h

```diff
@@ -349,7 +349,7 @@ class DLL_LINKAGE CGTownInstance : public CGDwelling, public IShipyard, public I
 
 	inline bool isBattleOutsideTown(const CGHeroInstance * defendingHero) const
 	{
-		return defendingHero && garrisonHero && defendingHero != garrisonHero;
```

Comment:
Are you sure about it? If garrisonHero is nullptr this must return false

Reply:
I'm not. but otherwise game crashes (https://github.com/vcmi/vcmi/issues/1024). That's sad if it's wrong because will be tough to fix that

Reply:
The idea is that if two heroes are in the town - garrison and visiting- first battle goes outside town but if not - inside town

Reply:
Even if visiting hero is not able to enter garrison because of extra units there but there is no hero - anyway battle goes as siege and in some cases these extra units just disappear.

Reply:
reverted